plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Purity.hs

```diff
@@ -50,11 +55,28 @@ data EvalTerm name uni fun a
   | EvalTerm Purity WorkFreedom (Term name uni fun a)
 
 instance
-  (PrettyBy config (Term name uni fun a))
-  => PrettyBy config (EvalTerm name uni fun a)
-  where
-  prettyBy _ Unknown                    = "<unknown>"
-  prettyBy config (EvalTerm eff work t) = pretty eff <+> pretty work <> ":" <+> prettyBy config t
+ ( Show name
+ , Everywhere uni Show
+ , Show fun
+ , Show a
+ , GShow uni
+ , Closed uni
+ ) => Show (EvalTerm name uni fun a) where
```

Comment:
Why do you need Show?

Reply:
For the unit test that compares eval terms: `assertEquals` needs `Eq` and `Show`